### DISCUSSION TITLE
Using GetText

### DISCUSSION THREAD

**Discussion by mdorkhah** [GateNLP/python-gatenlp#108] [Help Q&A]
May I know how to use GetText in order to get the text of matches?
The GetText(name="tok") gives me <gatenlp.pam.pampac.GetText at 0x180c5a58f10> but not the actual text.

**Comment by johann-petrak** [comment]
What do you want to achieve? How/where do you use GetText?  Could you share the code and context?
Note that GetText is a Getter helper class which can be used as part of an Action class to access the text of the match when an action is performed.

**Reply by mdorkhah** [reply]
Hi, I want to do something like this:
```
def gettext(succ, context, location):
    res = succ[0]
    new = None
    for data in res.matches:
        if data["name"] == "tok2":
            new = GetText("tok2")
            news.append(new)

p = Seq(Ann(type="New", name="tok1"), Ann(type="New", name="tok2"))
a = gettext
r = (p,a)
Pampac(r).run(doc, anns2match, outset) 
```

**Reply by johann-petrak** [reply]
Ah, no, with this you already have your own function that gets run at matching time. 

When you run `GetText("tok2")` this essentially only creates a callable (something like a function), but does not call that function which is conceptually the same as your function - declare when the rules are being declared, run at match time. 

You could instead create the GetText object first and then call it inside your function (untested, out of my head): 
```
textgetter = GetText("tok2", matchidx=0)
def gettext(succ, context, location):
    news.append(textgetter(succ, context, location)) 
```
This should get you the text of whatever the first (if there are more than one) match with the name "tok2" is.

If you want to get ANY/ALL matches with that name, doing it without GetText is probably easier:
```
textgetter = GetText("tok2", matchidx=0)
def gettext(succ, context, location):
    matches = succ[0].matches4name("tok2")   # get all matches with name "tok2" in the first/only result
    for match in matches:            # iterate over all matches (if any)
        span = match.get("span")    # get the offset range for the match, if there is one
        if span:
            news.append(doc[span])    # get the text for that offset range
```

**Reply by mdorkhah** [reply]
Thank you, it solved my problem.
